File: backend/src/main/scala/microservices/operations/airline/objects/ManagerFlightsPlannerRequest.scala
// 这个文件只定义 airline 管理员航班列表查询请求。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerFlightsPlannerRequest(
    managerId: String,
    managerType: String,
    departureAirports: Option[List[String]],
    arrivalAirports: Option[List[String]],
    departureDate: Option[String],
    timeRange: Option[String],
    sortDirection: Option[String]
)
object ManagerFlightsPlannerRequest:
  given sourceEncoder: Encoder[ManagerFlightsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ManagerFlightsPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/airline/objects/ManagerFlightsPlannerRequest.ts
// 本文件定义 ListManagerFlightsPlanner 的航空管理查询请求。

export type ManagerFlightsPlannerRequest = {
  managerId: string
  managerType: string
  departureAirports?: string[] | null
  arrivalAirports?: string[] | null
  departureDate?: string | null
  timeRange?: string | null
  sortDirection?: string | null
}

export const managerFlightsPlannerRequestFromJson = (json: string): ManagerFlightsPlannerRequest =>
  JSON.parse(json) as ManagerFlightsPlannerRequest

export const managerFlightsPlannerRequestToJson = (value: ManagerFlightsPlannerRequest): string =>
  JSON.stringify(value)
